admin/spam/views.py

```diff
@@ -37,47 +37,90 @@ def get_context_data(self, **kwargs):
         page_size = self.get_paginate_by(queryset)
         paginator, page, queryset, is_paginated = self.paginate_queryset(
             queryset, page_size)
-        context = {
+        return {
             'spam': map(serialize_comment, queryset),
             'page': page,
             'status': self.status,
             'page_number': page.number
         }
-        return super(SpamList, self).get_context_data(**context)
+
+
+class UserSpamList(SpamList):
+    template_name = 'users/spam.html'
+
+    def __init__(self):
+        self.user_id = None
+        super(UserSpamList, self).__init__()
+
+    def get_queryset(self):
+        self.status = self.request.GET.get('status', u'1')
+        self.user_id = self.kwargs.get('user_id', None)
+        query = (
+            Q('reports', 'ne', {}) &
+            Q('reports', 'ne', None) &
+            Q('user', 'eq', self.user_id) &
+            Q('spam_status', 'eq', int(self.status))
+        )
+        return Comment.find(query).sort(self.ordering)
+
+    def get_context_data(self, **kwargs):
+        queryset = kwargs.pop('object_list', self.object_list)
+        page_size = self.get_paginate_by(queryset)
+        paginator, page, queryset, is_paginated = self.paginate_queryset(
+            queryset, page_size)
+        return {
+            'spam': map(serialize_comment, queryset),
```

Comment:
I think this is dealt with in either #5045 or #5013. Though #5013 is obviously higher priority.